utensor_cgen/backend/utensor/code_generator/rearch/_code_generator.py

```diff
@@ -83,33 +91,62 @@ def _find_required_ops(self, ugraph):
         )
     return ops, placeholders, tensor_var_map
 
-  def _try_get_opt_snippets(self, ugraph, ops, tensor_var_map):
+  def _try_get_opt_snippets(self, ugraph, ops, tensor_var_map, placeholders):
+    # snippets for declaring ops
+    declare_op_snippets = []
+    construct_op_snippets = []
     # snippets to be rendered in the generated function body
     local_snippets = []
     # snippets to be rendered in the weight header file
     weight_snippets = []
+    # inputs/outputs enum names
+    input_enums = []
+    output_enums = []
     # op -> op variable name
     ops_map = {}
     for i, op in enumerate(ops):
-      op_var_name = 'op_{:03d}'.format(i)
+      op_var_name = 'op_{}_{:03d}'.format(op.op_type, i)
```

Comment:
```suggestion
      op_var_name = 'op_{}_{:03d}'.format(op.op_type.lower(), i)
```

Reply:
How about this so we can keep lower case as variable names, and upper case as class names :)

Reply:
Well, then write me the code that convert `'FullyConnectedOperator'` to `'fully_connected_operator'`.
It's not going to be simple and I think it's not necessary for codegen.

Reply:
Not to mention a tricker case: `Conv2dOpearator` -> `conv_2d_operator`

Reply:
The reason for such conversion is that `fullyconnectedoperator` is not a readable variable name to my eyes.

Reply:
I think just `conv2doperator` is fine for now. But as a challenge accepted
`re.sub(r"([A-Z]|[0-9]+[a-z]?)([a-z])", r"\1_\2", name)`